### PR TITLE
Add scoping for env vars check

### LINKED ISSUE #10448: [BuildCheck] EvaluationAnalysisScope configured in .editorconfig doesn't work properly
[BCTest_Repro.zip](https://github.com/user-attachments/files/16390266/BCTest_Repro.zip) has a solution with library projects, custom props and targets file in TestImport folder, as well as .editorconfig. Binary log is also included in the zip file. 

To reproduce, execute the steps below:
1.  Set environment variable `set Test=true`
2.  Build with analyzers: `dotnet <msbuild repo root>\artifacts\bin\bootstrap\net8.0\MSBuild\MSBuild.dll BCTest_Repro.sln -analyze -restore -bl`.
3.  Custom props file which is imported by ClassLibrary2.csproj accepting the environment variable `Test`. Because of  `build_check.BC0103.scope=project` configured in .editorconfig, BC0103 should not be triggered. Actual result is it's triggered and even applied to SDK imports.
4. Custom targets file copies a file to the solution folder. It's imported by ClassLibrary2.csproj and ClassLibrary3.csproj both. Because of  `build_check.BC0102.scope=project` configured in .editorconfig, BC0102 should not be triggered. Actual result is it's triggered for the imported target.
It looks like BuildCheck doesn't respect [EvaluationAnalysisScope documentation](https://github.com/dotnet/msbuild/blob/main/documentation/specs/BuildCheck/BuildCheck.md#scope-of-analysis).

### PR DESCRIPTION
**Pull request by JanKrivanek**
Fixes #10448

### Context
Added scoping implementation for the NoEnvVarsCheck

### Testing
Targetted test added - testing the scoping on imported file

### CODE REVIEW

File: src/Build/BuildCheck/Analyzers/NoEnvironmentVariablePropertyAnalyzer.cs [hunk 1]
```diff
@@ -26,13 +27,15 @@ internal sealed class NoEnvironmentVariablePropertyAnalyzer : BuildAnalyzer
...
     private bool _isVerboseEnvVarOutput;
+    private EvaluationAnalysisScope _scope;
 
     public override string FriendlyName => "MSBuild.NoEnvironmentVariablePropertyAnalyzer";
 
     public override IReadOnlyList<BuildAnalyzerRule> SupportedRules { get; } = [SupportedRule];
 
     public override void Initialize(ConfigurationContext configurationContext)
     {
+        _scope = configurationContext.BuildAnalyzerConfig[0].EvaluationAnalysisScope;
```
Comment (YuliiaKovalova) [inline src/Build/BuildCheck/Analyzers/NoEnvironmentVariablePropertyAnalyzer.cs:38]: can BuildAnalyzerConfig ever be null?

**Review by YuliiaKovalova — APPROVED** [review]

File: src/Build/BuildCheck/Analyzers/NoEnvironmentVariablePropertyAnalyzer.cs (hunk 1 above)
Comment (JanKrivanek) [reply src/Build/BuildCheck/Analyzers/NoEnvironmentVariablePropertyAnalyzer.cs:38]: No - that is the beauty of the nullability annotations :-)

But if you have asked if the array can be empty - then this cannot be enforced on the code analysis level, but still - our infra allways sets this to array of proper size (based on the rules count)

**Comment by YuliiaKovalova** [comment]
by the way, should we propagate scoping capabilities to bin log too?

File: src/BuildCheck.UnitTests/EndToEndTests.cs [hunk 2]
```diff
@@ -200,11 +201,51 @@ public void NoEnvironmentVariableProperty_Test(bool? customConfigEnabled, string
             customConfigData);
 
         string output = RunnerUtilities.ExecBootstrapedMSBuild(
-            $"{Path.GetFileName(projectFile.Path)} /m:1 -nr:False -restore -analyze", out bool success, false, _env.Output, timeoutMilliseconds: 120_000);
+            $"{Path.GetFileName(projectFile.Path)} /m:1 -nr:False -restore -analyze", out bool success, false, _env.Output);
```
Comment (AR-May) [inline src/BuildCheck.UnitTests/EndToEndTests.cs:204]: nit: Why to remove this timeout? Is some test running more that 2 minutes?

**Review by AR-May — APPROVED** [review]
LGTM

File: src/BuildCheck.UnitTests/EndToEndTests.cs (hunk 2 above)
Comment (JanKrivanek) [reply src/BuildCheck.UnitTests/EndToEndTests.cs]: The timeout is unnecessarily long. The default is 30 seconds and recently a high custom timeout started to be copy-paste spreaded over the code

**Pull request merged**